File: attribute-parser-compiler/src/main/java/pyxis/uzuki/live/attribute/parser/compiler/model/AttrDrawableModel.java
package pyxis.uzuki.live.attribute.parser.compiler.model;

import javax.lang.model.element.VariableElement;

import pyxis.uzuki.live.attribute.parser.annotation.AttrColor;
import pyxis.uzuki.live.attribute.parser.annotation.AttrDrawable;

/**
 * AttributesParser
 * Class: AttrIntModel
 * Created by Pyxis on 3/4/18.
 * <p>
 * Description:
 */

public class AttrDrawableModel extends BaseAttrModel {
    private String mSource;
    private int mDefValue;

    public AttrDrawableModel(VariableElement element) {
        super(element);

        AttrDrawable annotation = element.getAnnotation(AttrDrawable.class);

        mSource = annotation.value();
        mDefValue = annotation.def();

        findEnclosingClass();
    }

    public String getSource() {
        return mSource;
    }

    public int getDefValue() {
        return mDefValue;
    }
}

File: attribute-parser-compiler/src/main/java/pyxis/uzuki/live/attribute/parser/compiler/model/AttrDrawableModel.kt
package pyxis.uzuki.live.attribute.parser.compiler.model

import javax.lang.model.element.VariableElement

import pyxis.uzuki.live.attribute.parser.annotation.AttrColor
import pyxis.uzuki.live.attribute.parser.annotation.AttrDimension
import pyxis.uzuki.live.attribute.parser.annotation.AttrDrawable

/**
 * AttributesParser
 * Class: AttrIntModel
 * Created by Pyxis on 3/4/18.
 *
 *
 * Description:
 */

class AttrDrawableModel(element: VariableElement) : BaseAttrModel(element) {
    init {
        val annotation = element.getAnnotation<AttrDrawable>(AttrDrawable::class.java)
        source = annotation.value
    }
}